app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,59 @@
+module Katello
+  class RepositoryTypeManager
```

Comment:
I had originally suggested to use the foreman plugin models as a 'template' because

a) i really wasn't very aware of the bastion one
b) trying to avoid more katelloisms and ensure katello is as easy for foreman developers to work & look at

I tend to prefer the bastion model as well TBH